test/node/src/camel-case.test.ts

```diff
@@ -309,6 +309,93 @@ for (const dialect of DIALECTS) {
       })
     })
 
+    it('should respect underscoreBeforeDigits and not add a second underscore in a nested query', async () => {
+      interface CamelTestDatabase {
+        camelTest: CamelTest
+      }
+   
+      interface CamelTest {
+        id: Generated<number>
+        firstName: string
+        lastName: string
+        addressRow1: string
+        addressRow2: string
+      }
+   
+      let dbForMigration = new Kysely<CamelTestDatabase>({
+        ...ctx.config,
+        plugins: [new CamelCasePlugin()],
+      })
+   
+      await dbForMigration.schema.dropTable('test_camel').ifExists().execute()
+      await createTableWithId(dbForMigration.schema, dialect, 'test_camel')
+        .addColumn('first_name', 'varchar(255)')
+        .addColumn('last_name', 'varchar(255)')
+        .addColumn('address_row_1', 'varchar(255)')
+        .addColumn('address_row_2', 'varchar(255)')
+        .execute()
+   
+      let db = new Kysely<CamelTestDatabase>({
+        ...ctx.config,
+        plugins: [new CamelCasePlugin({ underscoreBeforeDigits: true })],
+      })
+   
+      if (dialect === 'mssql' || dialect === 'sqlite') {
+        db = db.withPlugin(new ParseJSONResultsPlugin())
+      }
```

Comment:
Let's reuse the existing setup as much as possible - adding necessary columns to existing interfaces and DDL queries.